week1/reposts/boccato/Main.hs

```diff
@@ -0,0 +1,38 @@
+module Main where
+
+import Control.Monad
+import Data.List
+import System.Random
+import Text.Printf
+
+nWinners = 7
+nSimulations = 10000
+nCurrentReposts = 1000000
+
+takeDistinct :: Eq a => Int -> [a] -> [a]
+takeDistinct n xs = sample n xs []
+  where
+    sample _ []     res  = res
+    sample n (x:xs) res
```

Comment:
It would be slightly easier to read if the function was written like this:

```
sample _ [] res = res
sample 0 _ res = res
sample n (x:xs) res
  | x `elem` res = sample ns xs res
  | otherwise = sample (n-1) xs (x:res)
```

because it doesn't conflate different cases together.